monai/apps/pathology/transforms/post/array.py

```diff
@@ -625,77 +633,150 @@ def __call__(  # type: ignore
-class HoVerNetNuclearTypePostProcessing(Transform):
+class HoVerNetPostProcessing(Transform):
     """
-    The whole post-procesing transform for nuclear type classification branch. It returns type prediction map and a
-    dict contains centroid, bounding box, type prediciton for each instance.
+    The post-processing transform for HoVerNet model, which includes all the three branches.
+    It generate a dictionary containing centroid, bounding box, type prediction for each instance. Also if requested,
+    it returns binary maps for instance segmentation and predicted types.
 
     Args:
-        min_num_points: assumed that the created contour does not form a contour if it does not contain more points
-            than the specified value. Defaults to 3.
-        level: optional. Used in `skimage.measure.find_contours`. Value along which to find contours in the array.
-            By default, the level is set to (max(image) + min(image)) / 2.
-        return_centroids: whether to return centroids for each instance.
-        output_classes: number of the nuclear type classes.
+        min_num_points: minimum number of points to be considered as a contour. Defaults to 3.
+        level: optional value for `skimage.measure.find_contours`, to find contours in the array.
+            If not provided, the level is set to (max(image) + min(image)) / 2.
+        distance_smooth_fn: smoothing function for distance map.
+            If not provided, :py:class:`monai.transforms.intensity.GaussianSmooth()` will be used..
+        marker_post_process_fn: post-process function for watershed markers.
+            If not provided, :py:class:`monai.transforms.post.FillHoles()` will be used.
+
     """
 
     def __init__(
         self,
         min_num_points: int = 3,
         level: Optional[float] = None,
-        return_centroids: Optional[bool] = False,
-        output_classes: Optional[int] = None,
+        distance_smooth_fn: Optional[Callable] = None,
+        marker_post_process_fn: Optional[Callable] = None,
     ) -> None:
         super().__init__()
-        self.return_centroids = return_centroids
-        self.output_classes = output_classes
 
+        if distance_smooth_fn is not None:
+            self.distance_smooth_fn = distance_smooth_fn
+        else:
+            self.distance_smooth_fn = GaussianSmooth()
+
+        if marker_post_process_fn is not None:
+            self.marker_post_process_fn = marker_post_process_fn
+        else:
+            self.marker_post_process_fn = FillHoles()
+
+        # instance segmentation transforms
+        self.generate_watershed_mask = GenerateWatershedMask(softmax=True)
+        self.generate_instance_border = GenerateInstanceBorder(kernel_size=3)
+        self.generate_distance_map = GenerateDistanceMap(smooth_fn=self.distance_smooth_fn)
+        self.generate_watershed_markers = GenerateWatershedMarkers(
+            threshold=0.7, radius=2, postprocess_fn=self.marker_post_process_fn
+        )
```

Comment:
Hi @drbeh, can l know why we still combine segmentation and classification post-processing? If do that, L673-L678 all of these transforms' parameters here are hard-coded and will be difficult for the user to modify. Another situation is that if the user wants to use their own `watershed` it will also become difficult to modify. Thanks!

Reply:
Hi @KumoLiu, thanks for your feedback. May I know why a user want to use their own `watershed`? This is a very specific post-processing for hovernet, how much of variance from this implementation do you think a user would expect?

Reply:
Apart from the fact that users use their own watershed, these parameters are definitely something users will want to change, right? What I want to say is actually not only to modify the `watershed`, I want to say for example to generate different distance maps or markers for the `watershed`. Maybe users want to do a comparison experiment or anything else. But in summary, doing it in such a way would cause less flexibility.

Reply:
Thanks, makes sense. I'll add more flexibility.